test/gtest/uct/test_md.cc

```diff
@@ -558,6 +558,39 @@ UCS_TEST_P(test_md, mem_query) {
     }
 }
 
+UCS_TEST_P(test_md, tl_resource_desc_v1_v2) {
+    uct_md_query_tl_resources_params_t params = {};
+    uct_tl_resource_desc_t *v1;
+    uct_tl_resource_desc_v2_t *v2;
+    unsigned num_v1, num_v2;
+    ucs_status_t status_v1, status_v2;
+
+    status_v1 = uct_md_query_tl_resources(md(), &v1, &num_v1);
+    status_v2 = uct_md_query_tl_resources_v2(md(), &v2, &num_v2, &params);
+
+    EXPECT_UCS_OK(status_v1);
+    EXPECT_UCS_OK(status_v2);
+    EXPECT_GT(num_v2, 0);
+    EXPECT_EQ(num_v2, num_v1);
+
+    for (auto i = 0; (i < num_v1) && (num_v1 == num_v2) ; ++i) {
```

Comment:
imo, no need to check `num_v1 == num_v2` again

Reply:
I think we need it because otherwise we might use out of bound memory.